### PR TITLE
Triggers referring to hidden question will produce an error.

### PR DESCRIPTION
**Pull request by gushil**
Closes #
#452 
#### Why is this the best possible solution? Were any other approaches considered?
The simplest one.
#### What are the regression risks?
None.
#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/XLSForm/xlsform.github.io) and include the link below.
None.
#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests_v1`
- [x] run `nosetests` and verified all tests pass
- [x] run `black pyxform` to format code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

**Comment by MartijnR** [comment]
Thanks Agus! I have not been able to break this.

File: pyxform/question.py [hunk 1]
```diff
@@ -32,6 +32,16 @@ def xml_instance(self, **kwargs):
 
     def xml_control(self):
         if ("calculate" in self.bind or self.trigger) and not (self.label or self.hint):
+            nested_setvalues = self.get_root().get_setvalues_for_question_name(
+                self.name
+            )
+            if nested_setvalues:
+                for setvalue in nested_setvalues:
+                    msg = (
+                        "Trigger was added for ${%s} that refers to hidden question. This is not allowed."
```
Comment (lognaturel) [inline pyxform/question.py:41]: I found this error message hard to parse. How about flipping it around so it's something like "${%s} is not a user-visible question so it can't be used as a calculation trigger for question ${%s}"? The first string would be the trigger (`self.name`) and the second would be the question that the `trigger` is set for (`setvalue[0]`). Hopefully I got that right.

I don't feel too strongly about it. If you prefer the current message, "to hidden question" -> "to a hidden question"

File: pyxform/tests_v1/test_trigger.py [hunk 2]
```diff
@@ -158,6 +158,22 @@ def test_handling_trigger_column_with_no_calculation_no_label_no_hint(self):
...
+            |        | calculate | one    |                      |         | 5 + 4       |
+            |        | calculate | one-ts |                      | ${one}  | now()       |
+            |        | note      | note   | timestamp: ${one-ts} |         |             |
+            """,
+            errored=True,
+            error__contains=[
+                "Trigger was added for ${one-ts} that refers to hidden question. This is not allowed."
+            ],
+        )
+
```
Comment (lognaturel) [inline pyxform/tests_v1/test_trigger.py:176]: Were you thinking that a test for a typed calculate is not necessary because `test_non_calculate_type_with_calculation_and_no_label_has_no_control` checks the branch in `question.py`? I'd still add one for completeness (`test_typed_calculate_cant_be_used_as_trigger` or something).
